sentry/src/main/java/io/sentry/clientreport/LockingClientReportStorage.java

```diff
@@ -0,0 +1,62 @@
+package io.sentry.clientreport;
+
+import io.sentry.ILogger;
+import io.sentry.SentryLevel;
+import io.sentry.SentryOptions;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+final class LockingClientReportStorage implements ClientReportStorage {
```

Comment:
I guess we can remove this, as it is slower than the Atomic version (6.5s vs 2s for 100M writes from 10 threads)

Reply:
What is slower, the add and reset methods?

Reply:
Mostly increasing the counts. But adding some resets didn't make a significant difference.

Reply:
I'm also in favor of only keeping one implementation of `ClientReportStorage`. We could then even remove the interface and only use the implementation.

Reply:
I'd keep the interface so a new storage can be added more easily. The current one only works well when there's a limited number of possible combinations. If the number of combinations grows we may want to have different implementations that prioritize lower memory usage over speed (and let the developer choose). We might even want different implementations for different android versions and the rest (since atomic update methods on ConcurrentHashMap are only available in newer API versions and other platforms).